src/test/groovy/com/example/CardServiceSpec.groovy

```diff
@@ -0,0 +1,60 @@
+package com.example
+
+import spock.lang.Specification
+
+class CardServiceSpec extends Specification {
+
+    def cardRepository = Stub(ListCardRepository)
+
+    def cardService = new CardService(cardRepository)
+
+    def "should return empty list of cards"() {
+        given:
+        cardRepository.findAll() >> { [] }
+
+        when:
+        def cards = cardService.findAll()
+
+        then:
+        cards.size() == 0
+    }
+
+    def "should return list with many elements"() {
+        given:
+        def firstCard = new Card("foo", Collections.singletonList("baz"))
+        def secondCard = new Card("bar", Collections.singletonList("baz"))
+        cardRepository.findAll() >> { [firstCard, secondCard] }
+
+        when:
+        def cards = cardService.findAll()
+
+        then:
+        cards as List == [secondCard, firstCard] as List
```

Comment:
A czemu jest `as List` a nie `as Set`? przecież to domyślnie jest lista i nie trzeba. Set był po to, żeby kolejności nie brać pod uwagę.